Icravideo

mehtank opened the pull request.

Commit:
Added min/max and 25/75 percentile ranges on v-plot

Commit:
Boxplots for looptime, fuel cost

Commit:
Bugfixed fillGaps

Commit:
More naive averaging, but works with stopped cars

Commit:
Added average looptime / fuel for all lanes aggregate

Commit:
More aggressive human lane changing

Commit:
List all vTypes included, even if count = 0

Commit:
ICRA sims

Commit:
Updated agents / functions

Commit:
getSpeedFactor for updated carFn

Commit:
Start in lane 0, add speedDev for humans

Commit:
Dunno made some changes that are long gone

Commit:
Restored actual sugiyama experiment

Commit:
Hack to plot variance of v over time

Commit:
Plot std dev of v

Commit:
New plots

Commit:
Better colorbar for single plots

Commit:
ACC vehicles too

Commit:
Added descriptive title to throwaway plots

Commit:
One lane sugiyama

Commit:
Removed obsolete code

Commit:
Additional config defaults

Commit:
Actually circular path

Commit:
Allow setting binary -> sumo-gui

Commit:
Allow setting of vehicle shape, and added defaults to agentTypes

Commit:
Color cars by speed

Commit:
Automatically export video frames with better coloring

cathywu commented on python/icra_double.py:
```diff
@@ -0,0 +1,74 @@
+import random
```
Should we get rid of icra_single and double? are they still useful outside of icra.py?

cathywu commented on python/loopsim.py:
```diff
@@ -69,22 +70,26 @@ def _mkdirs(self, name):
                 "-c", self.cfgfn,
-                "--remote-port", str(self.port)], 
+                "--remote-port", str(self.port)]
+        if sublane:
```
What is the sublane model used for?

cathywu commented on python/loopsim.py:
```diff
@@ -158,13 +190,18 @@ def _run(self, simSteps):
                 if carFn is not None:
                     carFn((idx, car), self, step)
+            if sumo == "sumo-gui":
+                traci.gui.screenshot("View #0", "%s/%08d.png" % (vid_path, step))
```
What does this do? (Can you add a comment?)

cathywu commented on python/loopsim.py:
```diff
@@ -158,13 +190,18 @@ def _run(self, simSteps):
                 car["x"] = self._getX(car["edge"], position)
                 car["v"] = traci.vehicle.getSpeed(v)
+                car["maxv"] = traci.vehicle.getMaxSpeed(v)
+                car["f"] = traci.vehicle.getSpeedFactor(v)
```
This doesn't change throughout the simulation, so can we retrieve and store this once?

cathywu commented on python/makecirc.py:
```diff
@@ -15,31 +18,45 @@ def makexml(name, nsl):
-    import subprocess
-    import sys
+def makenet(base, length, lanes, 
+        speedLimit=defaults.SPEED_LIMIT, 
```
nit: why are these now on new lines?

cathywu commented on python/makecirc.py:
```diff
@@ -145,8 +168,12 @@ def outputs(name, prefix="data/"):
         roufn=False
 
+    gui = E("viewsettings")
+    gui.append(E("scheme", name="real world"))
```
Beautiful.

cathywu commented on python/parsexml.py:
```diff
@@ -57,24 +58,48 @@ def parsexml(fn, edgestarts, xmax, vdefault=0):
 
         for lid, thislane in lanedata.iteritems():
-            f = interp([x["x"] for x in thislane], [x["v"] for x in thislane], xmax, vdefault)
+            vf = interp([x["x"] for x in thislane], [x["v"] for x in thislane], xmax, vdefault)
```
What does f mean? Frame? Can you add a comment?

cathywu commented on python/plots.py:
```diff
@@ -61,71 +61,128 @@ def pcolor(title, (xrng, xlabel),
-    ticks = linspace(smin, smax, 6)
+    fig.subplots_adjust(right=0.85)
+    if numlanes == 2:
+        cbm = 0.52
```
What are `cbm`, `cbl`?

Commit:
Removed code that's obsolete in icravideo [ but used for paper plots, so should still stay in icrasims ]

Commit:
Added comment

Commit:
Comment

Commit:
Comment

mehtank commented:
On Mon, Sep 19, 2016 at 05:36:25PM -0700, Cathy Wu wrote:

> cathywu commented on this pull request.
> 
> > @@ -0,0 +1,74 @@
> > +import random
> 
> Should we get rid of icra_single and double? are they still useful outside of icra.py?

Gone.  Should keep in icrasims branch though, because they're used to
make the paper graphics.

> > ```
> >              sumoBinary, 
> >              "--step-length", repr(self.simStepLength),
> >              "--no-step-log",
> >              "-c", self.cfgfn,
> > ```
> > -                "--remote-port", str(self.port)], 
> > -                "--remote-port", str(self.port)]
> > -        if sublane:
> 
> What is the sublane model used for?

It's meant to allow for smooth instead of instantaneous lane changes.
Turns out cars often just get stuck between lanes for some reason.
Probably need to set some parameters somewhere?

> > ```
> >              carFn = self.carFns[car["type"]]
> >              if carFn is not None:
> >                  carFn((idx, car), self, step)
> > ```
> > -            if sumo == "sumo-gui":
> > -                traci.gui.screenshot("View #0", "%s/%08d.png" % (vid_path, step))
> 
> What does this do? (Can you add a comment?)

Added.

> > @@ -158,13 +190,18 @@ def _run(self, simSteps):
> >                  car["lane"] = traci.vehicle.getLaneIndex(v)
> >                  car["x"] = self._getX(car["edge"], position)
> >                  car["v"] = traci.vehicle.getSpeed(v)
> > -                car["maxv"] = traci.vehicle.getMaxSpeed(v)
> > -                car["f"] = traci.vehicle.getSpeedFactor(v)
> 
> This doesn't change throughout the simulation, so can we retrieve and store this once?

Umm.  Yeah, we could.  There are probably other speedup tweaks that
would have a bigger effect though, like the sort() immediately
following.  I'm not sure it's worth doing yet.  

> > @@ -15,31 +18,45 @@ def makexml(name, nsl):
> >  def printxml(t, fn):
> >      etree.ElementTree(t).write(fn, pretty_print=True, encoding='UTF-8', xml_declaration=True) 
> 
> -def makenet(base, length, lanes, speedLimit=defaults.SPEED_LIMIT, path=""):
> -    import subprocess
> -    import sys
>   +def makenet(base, length, lanes, 
> -        speedLimit=defaults.SPEED_LIMIT, 
> 
> nit: why are these now on new lines?

It looked nicer to me that way.

> > @@ -145,8 +168,12 @@ def outputs(name, prefix="data/"):
> >      else:
> >          roufn=False
> -    gui = E("viewsettings")
> -    gui.append(E("scheme", name="real world"))
> 
> Beautiful.
> 
> > @@ -57,24 +58,48 @@ def parsexml(fn, edgestarts, xmax, vdefault=0):
> >              pass
> 
> ```
>      for lid, thislane in lanedata.iteritems():
> ```
> -            f = interp([x["x"] for x in thislane], [x["v"] for x in thislane], xmax, vdefault)
> -            vf = interp([x["x"] for x in thislane], [x["v"] for x in thislane], xmax, vdefault)
> 
> What does f mean? Frame? Can you add a comment?

f for function.  Comment added.

> > ```
> >  # Add colorbar
> > ```
> > -    fig.subplots_adjust(right=0.8)
> > -    cbar_ax = fig.add_axes([0.84, 0.35, 0.02, 0.55])
> > 
> > ## \-    cbar_ax2 = fig.add_axes([0.84, 0.1, 0.02, 0.2])
> > -    ticks = linspace(smin, smax, 6)
> > -    fig.subplots_adjust(right=0.85)
> > -    if numlanes == 2:
> > -        cbm = 0.52
> 
> What are `cbm`, `cbl`?

Random numbers to make the colorbar plot positioned in the right place.
Added comment.
